src/App.tsx

```diff
@@ -65,8 +229,16 @@ export const Main = () => {
       </header>
       <main className="flex-grow overflow-auto flex-col-reverse bg-body">
         <div className="px-4 py-4">
-          {data.posts.data.map((post: PostProps) => (
-            <PostItem {...post} />
+          {posts.map((post: PostProps, index: number) => (
+            <PostItem onClick={() => {
+
+              //Function for post selection used in Delete and update post functions
+              const postsArray = posts.map((post: PostProps) => {
```

Comment:
Which improvements we can apply to avoid iterating over the whole array on every click?
With big objects and with pagination this could create some problems.

Interesting in your opinion about this topic.